### PR TITLE
Stabilize operator for release and update docs

### PR DESCRIPTION
**Pull request by waveywaves**

### CODE REVIEW

File: controllers/jenkins_controller.go [hunk 1]
```diff
@@ -555,16 +563,6 @@ func (r *JenkinsReconciler) getCalculatedSpec(ctx context.Context, jenkins *v1al
 		calculatedSpec.Master.Containers = containers
 	}
 
-	if reflect.DeepEqual(calculatedSpec.JenkinsAPISettings, v1alpha2.JenkinsAPISettings{}) {
```
Comment (jkhelil) [inline controllers/jenkins_controller.go:558]: why are you removing this ? Please check with @akram as I think he introduced this

File: docs/backupandrestore.adoc [hunk 2]
```diff
@@ -0,0 +1,174 @@
+Backup and Restore
+-------------------
+
+You can backup and restore your Jenkins data related to jobs, plugin or configuration with the Backup and Restore
+feature provided by the Operator.
+
+There are CustomResourceDefinitions available which aid in trigerring the *Backup* and *Restore* given that backup has
```
Comment (jkhelil) [inline docs/backupandrestore.adoc]: I would rather make simple sentences:
For this, you need, first,  to enable the backup feature on the opeartor managed Jenkins instance, then, Use a backup or restore CustomResourceDefinition to create the backup or the restore.

File: docs/configurationascode.adoc [hunk 3]
```diff
@@ -1,22 +1,26 @@
...
 The role of the init container is to copy the configuration as code files you provide into a target configuration as code volume, mounted to the Jenkins container. This way we have a read write volume on the Jenkins container that holds your configuration as code files.
 Another Sidecar container is also created if you choose to enable autoreload of the configuration as code files. That way, if you change the configmaps for Jenkins configuration, The Jenkins configuration is reloaded.
 
-## Configure Jenkins
+Configure Jenkins
+~~~~~~~~~~~~~~~~~
 The Jenkins CR exposes attributes to configure Jenkins, you can specify the following in the Jenkins CR spec:
-> - enabled: speicifies if the configuration as code will be enabled
+> - enabled: specifies if the configuration as code will be enabled
 > - defaultConfig: specifies if you want the default config for Jenkins. See the the default configmap under deploy/default-config.yaml
```
Comment (jkhelil) [inline docs/configurationascode.adoc]: defaultConfig: specifies if you want the default config for Jenkins. 
"Delete this "
See the the default configmap under deploy/default-config.yaml

File: controllers/jenkins_controller.go (hunk 1 above)
Comment (waveywaves) [reply controllers/jenkins_controller.go:558]: I was under the impression that it was added in the legacy code.

Comment (waveywaves) [reply controllers/jenkins_controller.go:558]: Based on Akram's comment, we are removing this considering it's legacy

File: docs/configurationascode.adoc (hunk 3 above)
Comment (waveywaves) [reply docs/configurationascode.adoc]: updating

File: docs/backupandrestore.adoc (hunk 2 above)
Comment (waveywaves) [reply docs/backupandrestore.adoc]: I have updated as per your remark here. I will review and make it simple in the next PR if that's alright.

**Pull request merged**